mitodl/open-discussions
# Change is_subscriber to return correct result if the user is a subscriber but not a contributor to a private channel

#### What are the relevant tickets?
Fixes #187 

#### What's this PR do?
- Changes is_subscriber to look through the list of subscribed channels to see if a user is a subscriber. The channel info API will return 403 if the user is not a contributor so this is unreliable.
 - Handles double adds and deletes for moderator, subscriber and contributor. The status code should be 201 for an add and a 204 for a delete.

#### How should this be manually tested?
Make a private channel and have a user who is not on the list of contributors for that channel. Go to `/api/v0/channels/<channel>/subscribers/<username>/` and verify that it returns a 404. A DELETE to that API should return a 204, though it may not make a change since this is inaccessable in the reddit API.
